README.adoc

```diff
@@ -13,30 +29,273 @@ These can be installed via the provided `.tool-versions` using `asdf install`.
+
+> Note that, contrary to `errors`, `warnings` do not break testing or app compilation.
+
+You can manually lint, using:
+
+[source, shell]
+----
+yarn lint
+----
+
+or
+
+[source, shell]
+----
+yarn lint --fix
+----
+
+if you want to automatically fix linting issues.
+
+== Pre-push trigger
+
+We are using `Husky` to trigger the linting, testing and building of the app before pushing the code to GitHub to prevent unnecessary `Vercel` build.
+
+== TypeScript
+
+=== Type declaration
+
+The rule we follow is that, if a declared type is only used in one file, we locate it in said file. Otherwise, we move it in its own file, under `@types/`.
+The exceptions to this rule are:
+
+- *next-env.d.ts*: Adds NextJS types globally.
+- *assets.d.ts*: Allow the import of assets files in TypeScript files.
+
+as NextJS required them to be located at the root of the repository.
+
+=== Typing React components
+
+We chose to type React component like so:
+
+[source, typescript]
+----
+import React from "react";
+
+// Without props.
+export const Foo: React.FC = () => { return ;}
+
+// With props.
+type BarProps = { foo: "bar" }
+
+export const Bar: React.FC<BarProps> = ({ foo }) => { return <div>{ foo }</div>}
+----
+
+If you are not familiar with TypeScript generic types, please take a look at the https://www.typescriptlang.org/docs/handbook/generics.html[documentation].
+
+== Style
+
+=== Global style
+
+The `globalStyle` object, found in `/src/design-system/globals/globalStyle.tsx`, is used to remove undesired style and behavior found in HTML. 
+
+> Note that we chose to set the global font size to *62.5%*. This font size means that '1rem' is exactly equal to '10px', which makes setting REM values very simple.
+
+The `globalStyle` is scoped in the `_app` page component.
+
+=== Theme
+
+You should prioritize theme values over arbitrary CSS values.
+
+The theme structure is defined inside the `src/@types/styled-component.d.ts` definition type file.
+
+The theme values are set inside `src/design-system/theme/lightTheme.ts` file. Each field are set individually, then assigned to the `lightTheme` variable.
+
+The theme access is scoped in the `_app` page component.
+
+You can access the theme properties and values inside a `styled` component by calling the `theme` props:
+
+[source, typescript]
+----
+import styled from "styled-components";
+
+// Note that styled component are PascalCase.
+const StyledComponent = styled.div`
+  css-property: ${({ theme }) => theme.themeKey};
+`
+----
+
+You can access the theme properties and values from everywhere else with the custom `useTheme` hook:
+
+[source, typescript]
+----
+import { useTheme } from "relative/path/to/the/hook";
```

Comment:
```suggestion
import { useTheme } from "relative/path/to/the/hooks/useTheme";
```

where is it by the way?

Reply:
`src/design-system/theme/useTheme.ts`

Reply:
I'm still wondering if it should go into `/hooks` though.

Reply:
```suggestion
import { useTheme } from "relative/path/to/the/design-system/theme/useTheme";
```
then 😛